_source/_posts/2020-12-04-25-years-javascript-java.adoc

```diff
@@ -0,0 +1,156 @@
+
+The birthdate that many assign to Java -- May 23rd, 2020 -- was when the first public alpha release of the Java programming language and platform happened.
+
+Ben Evans published an excellent InfoQ article about https://www.infoq.com/news/2020/05/java-at-25/[Java at 25]. In it, he speaks the truth:
+
+> The Internet was not yet really a part of the mainstream of public consciousness.
+
+Oh my, how the times have changed!
+
+Java became popular in the late 90s with applets. Then it hit its stride in the early 2000s with server-side development. Struts became one of the first "trendy frameworks" in 2001-2002. If you had experience with it, you could get $100/hour. Or you could learn it on the job (like I did) and still make great money!
+
+Java enjoyed a glorious period from 2002-2006 with J2EE and all its web frameworks. Developers became frustrated with EJBs and Struts, and Hibernate and Spring rose to prominence as open source frameworks.
+
+image::{% asset_path 'blog/25-years-javascript-java/j2ee-without-ejb.jpg' %}[alt=J2EE without EJB,align=center]
+
+I had a lot of fun doing "Java web framework comparison" talks, starting in 2004. Then along came JavaScript with its renaissance.
+
+It's funny to look back and see that Flex 1.0, JSF 1.0, and Spring 1.0 were all released within weeks of each other. The term, Ajax, was coined in early 2005, and JavaScript eventually took out the first two, while Spring became a powerhouse on the server.
+
+image::{% asset_path 'blog/25-years-javascript-java/history-of-web-frameworks-timeline.png' %}[alt=History of Web Frameworks,align=center]
+
+NOTE: This timeline can be updated by creating a pull request to the https://github.com/mraible/history-of-web-frameworks-timeline[history-of-web-frameworks-timeline repository].
+
+Java didn't have much innovation while the JavaScript Renaissance was happening. I was a Java developer at LinkedIn in 2007-2008, then moved to full-time frontend development with GWT and jQuery at Evite in 2009.
+
+image:{% asset_path 'blog/25-years-javascript-java/spring-boot.png' %}[alt=Spring Boot,role="pull-right img-150px"]
+
+Java 8 was the next major thing to happen in JavaLand, and it wasn't released until March 18, 2014! A couple weeks later, Spring Boot 1.0 was released, on https://spring.io/blog/2014/04/01/spring-boot-1-0-ga-released[April Fools Day], no less!
+
+Spring Boot led to Spring Cloud and both helped fuel the microservices boom that continues today. MicroProfile was created to help with the stagnation in the Java EE world.
+
+Recently, there's been a lot of innovation happening with new frameworks like Micronaut and Quarkus. All the major server-side Java frameworks are _going native_ with GraalVM and making themselves relevant in a serverless, subatomic-startup-time, world.
```

Comment:
maybe add links for Micronaut and qurkaus?

Reply:
Done.